src/types.ts

```diff
@@ -1194,17 +1209,39 @@ export interface OpenAIAssistantDeleteParams extends AssistantDeleteParamsBase {
   config: OpenAIChatConfigureParams;
 }
 
-export interface Assistant {
+export type AssistantDeleteParams = OpenAIAssistantDeleteParams;
+
+export interface Thread {
   id: string;
+  messages: Array<ThreadMessage>;
+  metadata: object | null;
   created_at: number;
-  description: string | null;
-  instructions: string;
-  metadata: object;
-  model: string;
-  name: string;
 }
 
-export interface AssistantDeleted {
+export interface ThreadMessage {
   id: string;
-  deleted: boolean;
+  content: string;
+  role: 'user' | 'assistant';
+  metadata: object | null;
+  created_at: number;
+}
+
+interface ThreadCreateParamsBase {
+  tokenId: string;
+  messages?: Array<ThreadCreateParams.Message>;
+  metadata?: object | null;
```

Comment:
when creating thread and message, metadata is inputed and sent only to OpenAI, and is not save on the blockchain.